### ISSUE TITLE
Question: Please can someone help with this sample code from the tutorial?

### ISSUE THREAD

**Issue by david201275** [DotSpatial/DotSpatial#981]
Hi,
Really sorry to bother anyone with this but I am struggling. I have no previous experience of dotspatial.
I am working in VB.Net
I downloaded one of the tutorials from the codeplex website and the specific example shows how to draw points, lines and polygons and save them to a shp file.

The point example works fine but I cant get the line or polygon to work. 
The section of the tutorial Select Case statement code is below and I used nugget to install the dotspatial packages in my project. (v2.0.0-rc1)
It seems to be where I try to add coordinates that I have problems. I understand that the methods may be different for the v2.0 but I cant seem to find the correct way of doing it.

Dotspatial looks really interesting so any help would be much appreciated.

Thanks in advance.

```
Case "line"
                If e.Button = MouseButtons.Left Then
                    'left click - fill array of coordinates
                    'coordinate of clicked point
                    Dim coord As Coordinate = Map1.PixelToProj(e.Location)
                    If (linemouseClick) Then
                        'first time left click - create empty line feature
                        If firstClick Then
                            'Create a new List called lineArray.
                            'In List we need not define the size and also 
                            'Here this list will store the Coordinates
                            'We are going to store the mouse click coordinates into this array.
                            Dim lineArray As New List(Of Coordinate)
                            'Create an instance for LineString class.
                            'We need to pass collection of list coordinates
                            Dim lineGeometry As New LineString(lineArray)
                            'Add the linegeometry to line feature
                            Dim lineFeature As IFeature = lineF.AddFeature(lineGeometry)
                            'add first coordinate to the line feature
                            lineFeature.Coordinates.Add(coord)
                            'set the line feature attribute
                            lineID = lineID + 1
                            lineFeature.DataRow("LineID") = lineID
                            firstClick = False
                        Else
                            'second or more clicks - add points to the existing feature
                            Dim existingFeature As IFeature = lineF.Features(lineF.Features.Count - 1)
                            existingFeature.Coordinates.Add(coord)
                            'refresh the map if line has 2 or more points
                            If existingFeature.Coordinates.Count >= 2 Then
                                lineF.InitializeVertices()
                                Map1.ResetBuffer()
                            End If
                        End If
                    End If
                Else
                    'right click - reset first mouse click
                    firstClick = True
                    Map1.ResetBuffer()
                End If


            Case "polygon"
                If e.Button = MouseButtons.Left Then
                    'left click - fill array of coordinates
                    Dim coord As Coordinate = Map1.PixelToProj(e.Location)
                    If (polygonmouseClick) Then
                        'first time left click - create empty line feature
                        If firstClick Then
                            'Create a new List called polygonArray.
                            'Here this list will store the Coordinates
                            'We are going to store the mouse click coordinates into this array.
                            Dim polygonArray As New List(Of Coordinate)
                            'Create an instance for LinearRing class.
                            'We pass the polygon List to the constructor of this class
                            Dim polygonGeometry As New LinearRing(polygonArray)
                            'Add the polygonGeometry instance to PolygonFeature
                            Dim polygonFeature As IFeature = polygonF.AddFeature(polygonGeometry)
                            'add first coordinate to the polygon feature
                            polygonFeature.Coordinates.Add(coord)
                            'set the polygon feature attribute
                            polygonID = polygonID + 1
                            polygonFeature.DataRow("PolygonID") = polygonID
                            firstClick = False
                        Else
                            'second or more clicks - add points to the existing feature
                            Dim existingFeature As IFeature = polygonF.Features(polygonF.Features.Count - 1)
                            existingFeature.Coordinates.Add(coord)
                            'refresh the map if line has 2 or more points
                            If existingFeature.Coordinates.Count >= 3 Then
                                'refresh the map
                                polygonF.InitializeVertices()
                                Map1.ResetBuffer()
                            End If
                        End If
                    End If
                Else
                    'right click - reset first mouse click
                    firstClick = True
                    Map1.Cursor = Cursors.Arrow
                End If

```

**Comment by jany-tenaj** [comment]
> It seems to be where I try to add coordinates that I have problems. 

Please explain what problems you mean. Knowing what doesn't work helps fixing the problem. ^^

**Comment by david201275** [comment]
thanks for the response. I have included the errors that visual studio lists below in the error window.

- Value of type 'List(Of Coordinate)' cannot be converted to 'Coordinate()'.	
- 'Coordinates' is not a member of 'IFeature'

These errors are raised from the following lines in the sample code from the tutorial marked below with  '######## 

```
Case "line"
                If e.Button = MouseButtons.Left Then
                    'left click - fill array of coordinates
                    'coordinate of clicked point
                    Dim coord As Coordinate = Map1.PixelToProj(e.Location)
                    If (linemouseClick) Then
                        'first time left click - create empty line feature
                        If firstClick Then
                            'Create a new List called lineArray.
                            'In List we need not define the size and also 
                            'Here this list will store the Coordinates
                            'We are going to store the mouse click coordinates into this array.
                            Dim lineArray As New List(Of Coordinate)
                            'Create an instance for LineString class.
                            'We need to pass collection of list coordinates

                         '########    Dim lineGeometry As New LineString(lineArray)

                            'Add the linegeometry to line feature
                            Dim lineFeature As IFeature = lineF.AddFeature(lineGeometry)
                            'add first coordinate to the line feature

                            '########  lineFeature.Coordinates.Add(coord)

                            'set the line feature attribute
                            lineID = lineID + 1
                            lineFeature.DataRow("LineID") = lineID
                            firstClick = False
                        Else
                            'second or more clicks - add points to the existing feature
                            Dim existingFeature As IFeature = lineF.Features(lineF.Features.Count - 1)

                            '########  existingFeature.Coordinates.Add(coord)

                            'refresh the map if line has 2 or more points

                       '########      If  existingFeature.Coordinates.Count >= 2 Then

                                lineF.InitializeVertices()
                                Map1.ResetBuffer()
                            End If
                        End If
                    End If
                Else
                    'right click - reset first mouse click
                    firstClick = True
                    Map1.ResetBuffer()
                End If


            Case "polygon"
                If e.Button = MouseButtons.Left Then
                    'left click - fill array of coordinates
                    Dim coord As Coordinate = Map1.PixelToProj(e.Location)
                    If (polygonmouseClick) Then
                        'first time left click - create empty line feature
                        If firstClick Then
                            'Create a new List called polygonArray.
                            'Here this list will store the Coordinates
                            'We are going to store the mouse click coordinates into this array.
                            Dim polygonArray As New List(Of Coordinate)
                            'Create an instance for LinearRing class.
                            'We pass the polygon List to the constructor of this class

                            '########  Dim polygonGeometry As New LinearRing(polygonArray)

                            'Add the polygonGeometry instance to PolygonFeature
                            Dim polygonFeature As IFeature = polygonF.AddFeature(polygonGeometry)
                            'add first coordinate to the polygon feature

                            '######## polygonFeature.Coordinates.Add(coord)

                            'set the polygon feature attribute
                            polygonID = polygonID + 1
                            polygonFeature.DataRow("PolygonID") = polygonID
                            firstClick = False
                        Else
                            'second or more clicks - add points to the existing feature
                            Dim existingFeature As IFeature = polygonF.Features(polygonF.Features.Count - 1)

                         '########  existingFeature.Coordinates.Add(coord)

                            'refresh the map if line has 2 or more points

                          '########   If existingFeature.Coordinates.Count >= 3 Then

                                'refresh the map
                                polygonF.InitializeVertices()
                                Map1.ResetBuffer()
                            End If
                        End If
                    End If
                Else
                    'right click - reset first mouse click
                    firstClick = True
                    Map1.Cursor = Cursors.Arrow
                End If
```

**Comment by jany-tenaj** [comment]
>    Dim lineArray As New List(Of Coordinate)
>    Dim lineGeometry As New LineString(lineArray) '- Value of type 'List(Of Coordinate)' cannot be converted to 'Coordinate()'.	

This is self-explanatory. You're trying to use a list even though an array is expected.
Try this instead: `Dim lineGeometry As New LineString(lineArray.ToArray())`

You could replace 
> existingFeature.Coordinates.Add(coord)   - 'Coordinates' is not a member of 'IFeature'

by ` existingFeature.Geometry.Coordinates.Add(coord)` to get rid of the given error message. This is caused by the fact that in DS 2.0 it's no longer possible to use Feature for geometry based operations. You have to use Feature.Geometry instead.

But you'll cause the next error because Coordinates is an array and you can't add things to arrays.
So basically you have to change the way you work. 

First get all the points you want to use to create your geometry. Then add them to an array and use this array to create your geometry.



Be aware of issue #801.

**Comment by david201275** [comment]
Thank you so much for your help.
I really appreciate it and I think that has helped me to get around my issue.
